comp/dogstatsd/server/parse.go

```diff
@@ -38,7 +39,7 @@ var (
 // parser parses dogstatsd messages
 // not safe for concurent use
 type parser struct {
-	interner    *stringInterner
+	interner    *cache.KeyedInterner
```

Comment:
If we don't remove the old interner implementation, do we want instead to add an interface that will (for a time) let us choose between one implementation or the other ?

This will make sure all this code stays optional until we know it makes performances better

Reply:
We will want to provide a write-up of how this change impacts [low](https://github.com/DataDog/datadog-agent/tree/main/test/regression/cases/dogstatsd_string_interner_8MiB_100) and [high-end](https://github.com/DataDog/datadog-agent/tree/main/test/regression/cases/dogstatsd_string_interner_8MiB_100k) use cases, see [this notebook](https://app.datadoghq.com/notebook/6753069/agent-dogstatsd-stringinterner-investigation) as an example. If we are going to swap out the implementation like this I'd be interested in seeing the regression detector run here but I also think if the swap were configurable it'd be easier to rig experiments to understand the user implications of the work being proposed.

Reply:
If someone can pass me appropriate API keys (the back-to-back summits have made this tricky) for ddev, I can get the benchmark to start uploading data to the benchmark notebook.

Reply:
I can help you run this in the aml benchmark cluster, but I'd also like to see this run in the SMP regression detector.

once this passes enough of the lint/test/package build steps, the regression detector should run automatically on this PR.